<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="it">
<head>
<!-- Generated by javadoc (1.8.0_171) on Thu Aug 02 16:36:18 CEST 2018 -->
<title>Dataset</title>
<meta name="date" content="2018-08-02">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Dataset";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Dataset.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li><a href="../index-files/index-1.html">Index</a></li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?Datasets/Dataset.html" target="_top">Frames</a></li>
<li><a href="Dataset.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">Datasets</div>
<h2 title="Class Dataset" class="title">Class Dataset</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>Datasets.Dataset</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">Dataset</span>
extends java.lang.Object</pre>
<div class="block">Class to manage the input data set for the analysis
 Input format:
 numberOfSamples  numberOfGenes
 Gene_1 Gene_2 ... Gene_n
 SampleName_1 ValueOf_1(Gene_1) ValueOf_1(Gene_2) ... ValueOf_1(Gene_n)
 SampleName_2 ValueOf_2(Gene_1) ValueOf_2(Gene_2) ... ValueOf_2(Gene_n)
 ...
 SampleName_n ValueOf_n(Gene_1) ValueOf_n(Gene_2) ... ValueOf_n(Gene_n)</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>rossi</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#compacted">compacted</a></span></code>
<div class="block">if data set must be compacted (do not keep copies of the same genotype) or not</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#frequencies">frequencies</a></span></code>
<div class="block">number of times a given genotype is present in the dataset (always 1 if data set is not compacted)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;boolean[]&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#genotypes">genotypes</a></span></code>
<div class="block">genotypes present in the data set</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#initialized">initialized</a></span></code>
<div class="block">data set can read input only once</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#labels">labels</a></span></code>
<div class="block">Genes names</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#numberOfEntries">numberOfEntries</a></span></code>
<div class="block">number of genotypes present in the data set</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#Dataset--">Dataset</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#collapse-java.util.ArrayList-">collapse</a></span>(java.util.ArrayList&lt;<a href="../Utils/Triplet.html" title="class in Utils">Triplet</a>&lt;boolean[],java.lang.Integer,java.lang.Integer&gt;&gt;&nbsp;mutationCountAndFrequency)</code>
<div class="block">Removes the copies of the same genotype and increases the frequency of that genotype accordingly</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#compact--">compact</a></span>()</code>
<div class="block">Reduces entries of the data set by grouping equal genotypes and
 memorizing their frequency of occurrence</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;<a href="../Utils/Pair.html" title="class in Utils">Pair</a>&lt;boolean[],java.lang.Integer&gt;&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#countMutations--">countMutations</a></span>()</code>
<div class="block">counts number of mutations for each genotype</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#filterClonalGenotypes--">filterClonalGenotypes</a></span>()</code>
<div class="block">Eliminates non mutated genotypes present in the data set</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#frequencyOf-int-">frequencyOf</a></span>(int&nbsp;i)</code>&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>boolean[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#get-int-">get</a></span>(int&nbsp;i)</code>&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#getNumberOfGenes--">getNumberOfGenes</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#getNumberOfGenotypes--">getNumberOfGenotypes</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#normalizedFrequencyOf-int-">normalizedFrequencyOf</a></span>(int&nbsp;i)</code>&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#printBMLFormat--">printBMLFormat</a></span>()</code>
<div class="block">Print this data set in BML format</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#printTRONCOFormat--">printTRONCOFormat</a></span>()</code>
<div class="block">Print this data set in TRONCO format</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#read--">read</a></span>()</code>
<div class="block">read data set from STDIN</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#read-java.lang.String-">read</a></span>(java.lang.String&nbsp;path)</code>
<div class="block">read data set from a file</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#read-java.lang.String:A-boolean:A:A-">read</a></span>(java.lang.String[]&nbsp;labels,
    boolean[][]&nbsp;data)</code>
<div class="block">Initialize this data set manually</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#readData-java.util.Scanner-">readData</a></span>(java.util.Scanner&nbsp;in)</code>
<div class="block">effectively reads a data set from a stream</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#reconstructDataset-java.util.ArrayList-">reconstructDataset</a></span>(java.util.ArrayList&lt;<a href="../Utils/Triplet.html" title="class in Utils">Triplet</a>&lt;java.lang.Integer,java.lang.Integer,java.lang.String&gt;&gt;&nbsp;selectedGenes)</code>
<div class="block">reconstruct data set after having reduced the number of considered genes</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#shrink-int-">shrink</a></span>(int&nbsp;thres)</code>
<div class="block">Keeps in the dataset only the top mutated genes</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#toString--">toString</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>java.util.ArrayList&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../Datasets/Dataset.html#translate-boolean:A-">translate</a></span>(boolean[]&nbsp;genotype)</code>
<div class="block">Translates a genotype in its relative set of genes labels</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="initialized">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initialized</h4>
<pre>private&nbsp;boolean initialized</pre>
<div class="block">data set can read input only once</div>
</li>
</ul>
<a name="compacted">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compacted</h4>
<pre>private&nbsp;boolean compacted</pre>
<div class="block">if data set must be compacted (do not keep copies of the same genotype) or not</div>
</li>
</ul>
<a name="numberOfEntries">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>numberOfEntries</h4>
<pre>private&nbsp;int numberOfEntries</pre>
<div class="block">number of genotypes present in the data set</div>
</li>
</ul>
<a name="labels">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>labels</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.String&gt; labels</pre>
<div class="block">Genes names</div>
</li>
</ul>
<a name="genotypes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>genotypes</h4>
<pre>private&nbsp;java.util.ArrayList&lt;boolean[]&gt; genotypes</pre>
<div class="block">genotypes present in the data set</div>
</li>
</ul>
<a name="frequencies">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>frequencies</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.Integer&gt; frequencies</pre>
<div class="block">number of times a given genotype is present in the dataset (always 1 if data set is not compacted)</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="Dataset--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Dataset</h4>
<pre>public&nbsp;Dataset()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="read--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>public&nbsp;void&nbsp;read()
          throws <a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></pre>
<div class="block">read data set from STDIN</div>
<dl>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></code> - if the dataset was already initialized</dd>
</dl>
</li>
</ul>
<a name="read-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>public&nbsp;void&nbsp;read(java.lang.String&nbsp;path)
          throws <a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></pre>
<div class="block">read data set from a file</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>path</code> - to input file</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></code> - if the dataset was already initialized</dd>
</dl>
</li>
</ul>
<a name="readData-java.util.Scanner-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readData</h4>
<pre>private&nbsp;void&nbsp;readData(java.util.Scanner&nbsp;in)</pre>
<div class="block">effectively reads a data set from a stream</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>in</code> - input stream</dd>
</dl>
</li>
</ul>
<a name="read-java.lang.String:A-boolean:A:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>public&nbsp;void&nbsp;read(java.lang.String[]&nbsp;labels,
                 boolean[][]&nbsp;data)
          throws <a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></pre>
<div class="block">Initialize this data set manually</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>labels</code> - for genes</dd>
<dd><code>data</code> - boolean matrix representing genotypes</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets">DatasetAlreadyInitializedException</a></code> - if the dataset was already initialized</dd>
</dl>
</li>
</ul>
<a name="compact--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compact</h4>
<pre>public&nbsp;void&nbsp;compact()</pre>
<div class="block">Reduces entries of the data set by grouping equal genotypes and
 memorizing their frequency of occurrence</div>
</li>
</ul>
<a name="filterClonalGenotypes--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>filterClonalGenotypes</h4>
<pre>private&nbsp;void&nbsp;filterClonalGenotypes()</pre>
<div class="block">Eliminates non mutated genotypes present in the data set</div>
</li>
</ul>
<a name="shrink-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>shrink</h4>
<pre>public&nbsp;void&nbsp;shrink(int&nbsp;thres)
            throws <a href="../Datasets/InvalidThresholdException.html" title="class in Datasets">InvalidThresholdException</a></pre>
<div class="block">Keeps in the dataset only the top mutated genes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>thres</code> - number of genes to be kept (increased in case of parity of occurrences</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/InvalidThresholdException.html" title="class in Datasets">InvalidThresholdException</a></code> - if thres is not a positive number</dd>
</dl>
</li>
</ul>
<a name="reconstructDataset-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reconstructDataset</h4>
<pre>private&nbsp;void&nbsp;reconstructDataset(java.util.ArrayList&lt;<a href="../Utils/Triplet.html" title="class in Utils">Triplet</a>&lt;java.lang.Integer,java.lang.Integer,java.lang.String&gt;&gt;&nbsp;selectedGenes)</pre>
<div class="block">reconstruct data set after having reduced the number of considered genes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>selectedGenes</code> - genotypes</dd>
</dl>
</li>
</ul>
<a name="collapse-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>collapse</h4>
<pre>private&nbsp;void&nbsp;collapse(java.util.ArrayList&lt;<a href="../Utils/Triplet.html" title="class in Utils">Triplet</a>&lt;boolean[],java.lang.Integer,java.lang.Integer&gt;&gt;&nbsp;mutationCountAndFrequency)</pre>
<div class="block">Removes the copies of the same genotype and increases the frequency of that genotype accordingly</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>mutationCountAndFrequency</code> - genotypes REQUIRE sorted by binary number and number of mutations</dd>
</dl>
</li>
</ul>
<a name="countMutations--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>countMutations</h4>
<pre>private&nbsp;java.util.ArrayList&lt;<a href="../Utils/Pair.html" title="class in Utils">Pair</a>&lt;boolean[],java.lang.Integer&gt;&gt;&nbsp;countMutations()</pre>
<div class="block">counts number of mutations for each genotype</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>genotypes associated with the number of mutations they contain</dd>
</dl>
</li>
</ul>
<a name="normalizedFrequencyOf-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>normalizedFrequencyOf</h4>
<pre>public&nbsp;double&nbsp;normalizedFrequencyOf(int&nbsp;i)
                             throws <a href="../Datasets/NotAlreadyCompactedException.html" title="class in Datasets">NotAlreadyCompactedException</a></pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>i</code> - genotype position</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>observed probability of that genotype</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/NotAlreadyCompactedException.html" title="class in Datasets">NotAlreadyCompactedException</a></code> - when this is not compacted</dd>
</dl>
</li>
</ul>
<a name="frequencyOf-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>frequencyOf</h4>
<pre>public&nbsp;int&nbsp;frequencyOf(int&nbsp;i)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>i</code> - genotype position</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>observed frequency of that genotype</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../Datasets/NotAlreadyCompactedException.html" title="class in Datasets">NotAlreadyCompactedException</a></code> - when this is not compacted</dd>
</dl>
</li>
</ul>
<a name="get-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>get</h4>
<pre>public&nbsp;boolean[]&nbsp;get(int&nbsp;i)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>i</code> - position</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>genotype at position 'i'</dd>
</dl>
</li>
</ul>
<a name="getNumberOfGenotypes--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfGenotypes</h4>
<pre>public&nbsp;int&nbsp;getNumberOfGenotypes()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of genotypes in this dataset</dd>
</dl>
</li>
</ul>
<a name="toString--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;java.lang.String&nbsp;toString()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a name="translate-boolean:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>translate</h4>
<pre>public&nbsp;java.util.ArrayList&lt;java.lang.String&gt;&nbsp;translate(boolean[]&nbsp;genotype)</pre>
<div class="block">Translates a genotype in its relative set of genes labels</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genotype</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a set of labels</dd>
</dl>
</li>
</ul>
<a name="getNumberOfGenes--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNumberOfGenes</h4>
<pre>public&nbsp;int&nbsp;getNumberOfGenes()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of genes considered in this dataset</dd>
</dl>
</li>
</ul>
<a name="printBMLFormat--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>printBMLFormat</h4>
<pre>public&nbsp;void&nbsp;printBMLFormat()</pre>
<div class="block">Print this data set in BML format</div>
</li>
</ul>
<a name="printTRONCOFormat--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>printTRONCOFormat</h4>
<pre>public&nbsp;void&nbsp;printTRONCOFormat()</pre>
<div class="block">Print this data set in TRONCO format</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Dataset.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li><a href="../index-files/index-1.html">Index</a></li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../Datasets/DatasetAlreadyInitializedException.html" title="class in Datasets"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?Datasets/Dataset.html" target="_top">Frames</a></li>
<li><a href="Dataset.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
