<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="it">
<head>
<!-- Generated by javadoc (1.8.0_171) on Sun Jul 15 14:41:34 CEST 2018 -->
<title>GenotypeGraph</title>
<meta name="date" content="2018-07-15">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<script type="text/javascript" src="../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenotypeGraph";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenotypeGraph.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li><a href="../index-files/index-1.html">Index</a></li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../GenotypeGraph/GenotypeGraphSimple.html" title="class in GenotypeGraph"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?GenotypeGraph/GenotypeGraph.html" target="_top">Frames</a></li>
<li><a href="GenotypeGraph.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">GenotypeGraph</div>
<h2 title="Class GenotypeGraph" class="title">Class GenotypeGraph</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>GenotypeGraph.GenotypeGraph</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../GenotypeGraph/GenotypeGraphSimple.html" title="class in GenotypeGraph">GenotypeGraphSimple</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="typeNameLabel">GenotypeGraph</span>
extends java.lang.Object</pre>
<div class="block">Base class to represent directed graphs about genotypes
 It also implements a basic dot format output 
 The implementation is with adjacency lists</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>rossi</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>(package private) java.lang.String[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#geneLabelsOrder">geneLabelsOrder</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>(package private) int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#id">id</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>(package private) java.util.ArrayList&lt;<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#nodes">nodes</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>private int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#threshold">threshold</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#GenotypeGraph--">GenotypeGraph</a></span>()</code>
<div class="block">Alternate constructor reading input from STDIN</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#GenotypeGraph-int-">GenotypeGraph</a></span>(int&nbsp;thres)</code>
<div class="block">Alternate constructor reading input from STDIN and 
 setting a threshold on the maximum number of genes</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#GenotypeGraph-java.lang.String:A-java.util.ArrayList-">GenotypeGraph</a></span>(java.lang.String[]&nbsp;labels,
             java.util.ArrayList&lt;boolean[]&gt;&nbsp;genotypes)</code>
<div class="block">default constructor, creates a graph consisting
 of many genotype nodes</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#GenotypeGraph-java.lang.String:A-java.util.ArrayList-int-">GenotypeGraph</a></span>(java.lang.String[]&nbsp;labels,
             java.util.ArrayList&lt;boolean[]&gt;&nbsp;genotypes,
             int&nbsp;thres)</code>
<div class="block">Alternate constructor also setting a threshold on the maximum number of genes</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>private boolean[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#computeElimiations-java.util.ArrayList-">computeElimiations</a></span>(java.util.ArrayList&lt;java.lang.Integer&gt;&nbsp;geneMutations)</code>
<div class="block">Finds the genes that should be considered according to the threshold set by the user</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>private java.util.ArrayList&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#computeGeneMutations-java.util.ArrayList-">computeGeneMutations</a></span>(java.util.ArrayList&lt;boolean[]&gt;&nbsp;tempSamples)</code>
<div class="block">Computes the array of the counts for each mutation
 (i.e. how many times every mutation was seen in any sample)</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>java.lang.String[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#getLabels--">getLabels</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#nodeLabel-GenotypeGraph.GenotypeNode-">nodeLabel</a></span>(<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&nbsp;node)</code>
<div class="block">Prints a node referring to mutated genes with their respective HUGO symbols</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#prepareEdges--">prepareEdges</a></span>()</code>
<div class="block">prepares the edges of the graph</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#readInput--">readInput</a></span>()</code>
<div class="block">reads an input file in "BML" format from STDIN:
 #samples       #genes '\n'
 spaced list of all HUGO symbols for each gene '\n'
 sampleID and relative boolean array of mutation presence '\n' (repeated #samples times)</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>private void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#reduceGenes-boolean:A-java.util.ArrayList-">reduceGenes</a></span>(boolean[]&nbsp;toBeKept,
           java.util.ArrayList&lt;boolean[]&gt;&nbsp;tempSamples)</code>
<div class="block">Applies the threshold set by the user to reduce the dataset</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#setNodes-java.util.ArrayList-">setNodes</a></span>(java.util.ArrayList&lt;<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&gt;&nbsp;newNodes)</code>
<div class="block">updates the node set to a new one
 (also dot IDs are recomputed)</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#size--">size</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>private boolean[][]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#toBoolMatrix-java.util.ArrayList-">toBoolMatrix</a></span>(java.util.ArrayList&lt;boolean[]&gt;&nbsp;list)</code>
<div class="block">Casts an ArrayList<boolean[]> to a boolean[][]</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#toDot--">toDot</a></span>()</code>
<div class="block">print this graph in dot format</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../GenotypeGraph/GenotypeGraph.html#toDotNode-GenotypeGraph.GenotypeNode-">toDotNode</a></span>(<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&nbsp;node)</code>
<div class="block">prints a node in dot format</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="geneLabelsOrder">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>geneLabelsOrder</h4>
<pre>java.lang.String[] geneLabelsOrder</pre>
</li>
</ul>
<a name="nodes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nodes</h4>
<pre>java.util.ArrayList&lt;<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&gt; nodes</pre>
</li>
</ul>
<a name="id">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>id</h4>
<pre>int id</pre>
</li>
</ul>
<a name="threshold">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>threshold</h4>
<pre>private&nbsp;int threshold</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenotypeGraph-java.lang.String:A-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenotypeGraph</h4>
<pre>public&nbsp;GenotypeGraph(java.lang.String[]&nbsp;labels,
                     java.util.ArrayList&lt;boolean[]&gt;&nbsp;genotypes)</pre>
<div class="block">default constructor, creates a graph consisting
 of many genotype nodes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>labels</code> - list associating each gene's HUGO symbol with a position</dd>
<dd><code>genotypes</code> - note that a single genotype is a sequence of boolean
 stating if a gene was or wasn't mutated (in the order defined by labels)</dd>
</dl>
</li>
</ul>
<a name="GenotypeGraph-java.lang.String:A-java.util.ArrayList-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenotypeGraph</h4>
<pre>public&nbsp;GenotypeGraph(java.lang.String[]&nbsp;labels,
                     java.util.ArrayList&lt;boolean[]&gt;&nbsp;genotypes,
                     int&nbsp;thres)</pre>
<div class="block">Alternate constructor also setting a threshold on the maximum number of genes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>labels</code> - </dd>
<dd><code>genotypes</code> - </dd>
</dl>
</li>
</ul>
<a name="GenotypeGraph--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>GenotypeGraph</h4>
<pre>public&nbsp;GenotypeGraph()</pre>
<div class="block">Alternate constructor reading input from STDIN</div>
</li>
</ul>
<a name="GenotypeGraph-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenotypeGraph</h4>
<pre>public&nbsp;GenotypeGraph(int&nbsp;thres)</pre>
<div class="block">Alternate constructor reading input from STDIN and 
 setting a threshold on the maximum number of genes</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>thres</code> - number of genes to keep for the analysis 
 (it selects the most mutated genes)</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getLabels--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLabels</h4>
<pre>public&nbsp;java.lang.String[]&nbsp;getLabels()</pre>
</li>
</ul>
<a name="readInput--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readInput</h4>
<pre>private&nbsp;void&nbsp;readInput()</pre>
<div class="block">reads an input file in "BML" format from STDIN:
 #samples       #genes '\n'
 spaced list of all HUGO symbols for each gene '\n'
 sampleID and relative boolean array of mutation presence '\n' (repeated #samples times)</div>
</li>
</ul>
<a name="reduceGenes-boolean:A-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reduceGenes</h4>
<pre>private&nbsp;void&nbsp;reduceGenes(boolean[]&nbsp;toBeKept,
                         java.util.ArrayList&lt;boolean[]&gt;&nbsp;tempSamples)</pre>
<div class="block">Applies the threshold set by the user to reduce the dataset</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>toBeKept</code> - list of genes to keep an to eliminate (in the first GeneLabelsOrder order)</dd>
<dd><code>tempSamples</code> - original unreduced samples</dd>
</dl>
</li>
</ul>
<a name="toBoolMatrix-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toBoolMatrix</h4>
<pre>private&nbsp;boolean[][]&nbsp;toBoolMatrix(java.util.ArrayList&lt;boolean[]&gt;&nbsp;list)</pre>
<div class="block">Casts an ArrayList<boolean[]> to a boolean[][]</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>list</code> - (of arrays)</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a matrix</dd>
</dl>
</li>
</ul>
<a name="computeElimiations-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeElimiations</h4>
<pre>private&nbsp;boolean[]&nbsp;computeElimiations(java.util.ArrayList&lt;java.lang.Integer&gt;&nbsp;geneMutations)</pre>
<div class="block">Finds the genes that should be considered according to the threshold set by the user</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>geneMutations</code> - for each gene (in the GeneLabelsOrder order) how many times it is found as mutated in the data set</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>list of genes to keep an to eliminate (in the GeneLabelsOrder order)</dd>
</dl>
</li>
</ul>
<a name="computeGeneMutations-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeGeneMutations</h4>
<pre>private&nbsp;java.util.ArrayList&lt;java.lang.Integer&gt;&nbsp;computeGeneMutations(java.util.ArrayList&lt;boolean[]&gt;&nbsp;tempSamples)</pre>
<div class="block">Computes the array of the counts for each mutation
 (i.e. how many times every mutation was seen in any sample)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tempSamples</code> - input dataset</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>array of the counts for each mutation</dd>
</dl>
</li>
</ul>
<a name="prepareEdges--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>prepareEdges</h4>
<pre>protected&nbsp;void&nbsp;prepareEdges()</pre>
<div class="block">prepares the edges of the graph</div>
</li>
</ul>
<a name="size--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>size</h4>
<pre>public&nbsp;long&nbsp;size()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of nodes of the graph</dd>
</dl>
</li>
</ul>
<a name="toDot--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toDot</h4>
<pre>public&nbsp;void&nbsp;toDot()</pre>
<div class="block">print this graph in dot format</div>
</li>
</ul>
<a name="toDotNode-GenotypeGraph.GenotypeNode-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toDotNode</h4>
<pre>public&nbsp;void&nbsp;toDotNode(<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&nbsp;node)</pre>
<div class="block">prints a node in dot format</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>node</code> - the node to be printed REQUIRE not null</dd>
</dl>
</li>
</ul>
<a name="nodeLabel-GenotypeGraph.GenotypeNode-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nodeLabel</h4>
<pre>public&nbsp;java.lang.String&nbsp;nodeLabel(<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&nbsp;node)</pre>
<div class="block">Prints a node referring to mutated genes with their respective HUGO symbols</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>node</code> - the node of which the genotype should be printed REQUIRE not null</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the sequence of HUGO symbols of genes that are mutated in a given genotype</dd>
</dl>
</li>
</ul>
<a name="setNodes-java.util.ArrayList-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setNodes</h4>
<pre>protected&nbsp;void&nbsp;setNodes(java.util.ArrayList&lt;<a href="../GenotypeGraph/GenotypeNode.html" title="class in GenotypeGraph">GenotypeNode</a>&gt;&nbsp;newNodes)</pre>
<div class="block">updates the node set to a new one
 (also dot IDs are recomputed)</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>newNodes</code> - new node set</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenotypeGraph.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li><a href="../index-files/index-1.html">Index</a></li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev&nbsp;Class</li>
<li><a href="../GenotypeGraph/GenotypeGraphSimple.html" title="class in GenotypeGraph"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?GenotypeGraph/GenotypeGraph.html" target="_top">Frames</a></li>
<li><a href="GenotypeGraph.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
